### PR TITLE
Add `with_***` helpers

### PR DESCRIPTION
**Pull request by soutaro**
This PR moves the helpers like `with_int` from `test/stdlib` to `lib/rbs/unit_test`, so that other gems can use them to test their type definitions.

### CODE REVIEW

File: lib/rbs/unit_test/with_aliases.rb [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      end
+
+      def with_array(*elements)
+        return WithEnum.new to_enum(__method__ || raise, *elements) unless block_given?
+
+        yield elements
+        yield ToArray.new(*elements)
+      end
+
+      def with_hash(hash = {})
```
Comment (ParadoxV5) [inline lib/rbs/unit_test/with_aliases.rb:71]: `with_array(*elements)`, but not `with_hash(**elements)`?

Note: kwargs isn’t Symbols only and that – good?
```ruby
p('any key' => 'is valid', 420 => 69)
  #=> {"any key"=>"is valid", 420=>69}
```

File: test/stdlib/test_helper.rb [hunk 2]
```diff
@@ -411,15 +106,33 @@ def each(&block)
...
+ToC = RBS::UnitTest::Convertibles::ToC
+ToStr = RBS::UnitTest::Convertibles::ToStr
+ToS = RBS::UnitTest::Convertibles::ToS
+ToSym = RBS::UnitTest::Convertibles::ToSym
+ToA = RBS::UnitTest::Convertibles::ToA
+ToArray = RBS::UnitTest::Convertibles::ToArray
+ToHash = RBS::UnitTest::Convertibles::ToHash
+ToPath = RBS::UnitTest::Convertibles::ToPath
+CustomRange = RBS::UnitTest::Convertibles::CustomRange
+Each = RBS::UnitTest::Convertibles::Each
```
Comment (ParadoxV5) [inline test/stdlib/test_helper.rb:130]: `include RBS::UnitTest::Convertibles`?

Comment (soutaro) [reply test/stdlib/test_helper.rb:130]: Ah this is in fact for RBS runtime type checker. I'm not sure if including works for it...

File: lib/rbs/unit_test/with_aliases.rb (hunk 1 above)
Comment (soutaro) [reply lib/rbs/unit_test/with_aliases.rb:71]: We can pass hash without curly brace with the current definition. So any other good reason to make it kwargs?

Comment (ParadoxV5) [reply lib/rbs/unit_test/with_aliases.rb:71]: I just prefer consistency.

**Comment by ksss** [comment]
I see potential for property based testing in this framework.
Do you have any plans for such an extension in the future?

```rb
with_int(count: 100) do |i|
  # Yields 100 times random integer value
end
```

File: lib/rbs/unit_test/convertibles.rb [hunk 3]
```diff
@@ -0,0 +1,176 @@
+# frozen_string_literal: true
+
+module RBS
+  module UnitTest
+    module Convertibles
+      class BlankSlate < BasicObject
+        instance_methods.each do |im|
+          next if %i[__send__ __id__].include? im
```
Comment (sampersand) [inline lib/rbs/unit_test/convertibles.rb:8]: We might want to remove `__id__` from this list, as it doesnt warn when you `undef` it

File: lib/rbs/unit_test/with_aliases.rb [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+      class WithEnum
+        include Enumerable
+
+        def initialize(enum) = @enum = enum
+
+        def each(&block) = @enum.each(&block)
+
+        def and_nil(&block)
+          self.and(nil, &_ = block)
```
Comment (sampersand) [inline lib/rbs/unit_test/with_aliases.rb:16]: We should add `return WithEnum.new to_enum(__method__ || raise) unless block` here

File: sig/unit_test/convertibles.rbs [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+      #
+      class BlankSlate < BasicObject
+        # Add Object methods to `self`
+        #
+        def __with_object_methods: (*Symbol methods) -> self
+      end
+
+      # An object with `#to_io` method
+      class ToIO < BlankSlate
+        @io: untyped
```
Comment (sampersand) [inline sig/unit_test/convertibles.rbs:16]: Shouldn't these `untyped`s be given types? ditto for `initialize` and `to_xxx` for each of the `ToXXX` classes

File: sig/unit_test/with_aliases.rbs [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+                    | (?Float value) -> WithEnum[float]
+
+      # Yields `::string` objects
+      #
+      def with_string: (?String value) { (string) -> void } -> void
+                     | (?String value) -> WithEnum[string]
+
+      # Yields `::array` objects
+      #
+      def with_array: (*untyped elements) { (array[untyped]) -> void } -> void
```
Comment (sampersand) [inline sig/unit_test/with_aliases.rbs:88]: We should give this a type, like `def with_array: [T] (*T elements)`. Same with `hash` and `range`

File: test/stdlib/File_test.rb
Comment (sampersand) [inline test/stdlib/File_test.rb:1]: unintentional inclusion?

Comment (soutaro) [reply test/stdlib/File_test.rb:1]: Yeah, it's unrelated, but I think it's acceptable to mix it.

File: lib/rbs/unit_test/with_aliases.rb (hunk 4 above)
Comment (soutaro) [reply lib/rbs/unit_test/with_aliases.rb:16]: and accepts no block call. So it should be ok.

File: lib/rbs/unit_test/convertibles.rb (hunk 3 above)
Comment (soutaro) [reply lib/rbs/unit_test/convertibles.rb:8]: Hmm. Will try it.

File: sig/unit_test/convertibles.rbs (hunk 5 above)
Comment (soutaro) [reply sig/unit_test/convertibles.rbs:16]: What is the type of IO here? I think it's not very clear and decided to keep them untyped.

File: sig/unit_test/with_aliases.rbs (hunk 6 above)
Comment (soutaro) [reply sig/unit_test/with_aliases.rbs:88]: Skip `range` method because it requires `<=>` method.

File: sig/unit_test/convertibles.rbs (hunk 5 above)
Comment (sampersand) [reply sig/unit_test/convertibles.rbs:16]: `@io` should be `::IO`; Same for `ToStr` and `String`, etc

**Pull request merged**